devito/passes/clusters/asynchrony.py

```diff
@@ -257,7 +258,7 @@ def _actions_from_update_memcpy(self, cluster, clusters, prefix, actions):
 
         # If fetching into e.g. `ub[sb1]` we'll need to prefetch into e.g. `ub[sb0]`
         tindex0 = e.lhs.indices[d]
-        if is_integer(tindex0):
+        if is_integer(tindex0) or isinstance(tindex0, IntDiv):
```

Comment:
Maybe add `or isintance(tindex0, Mod)` for case someone use % instead of IntDiv

Reply:
for that you end up in the `else`, deliberately :) 

what you see here isn't stuff that the user has written, but rather by-product expressions of other passes such as buffering